.github/workflows/tests-future-deps.yml

```diff
@@ -0,0 +1,165 @@
+          python -m pip install -i https://pypi.anaconda.org/scipy-wheels-nightly/simple numpy
+
+          # Install dependencies from PyPI.
+          python -m pip install --upgrade $PRE \
+            cycler fonttools kiwisolver packaging pillow pyparsing \
+            python-dateutil setuptools-scm \
+            -r requirements/testing/all.txt \
+            -r requirements/testing/extra.txt
+
+          # Install optional dependencies from PyPI.
+          # Sphinx is needed to run sphinxext tests
+          python -m pip install --upgrade sphinx
+
+          # GUI toolkits are pip-installable only for some versions of Python
+          # so don't fail if we can't install them.  Make it easier to check
+          # whether the install was successful by trying to import the toolkit
+          # (sometimes, the install appears to be successful but shared
+          # libraries cannot be loaded at runtime, so an actual import is a
+          # better check).
+          # PyGObject, pycairo, and cariocffi do not install on OSX 10.12.
+           python -m pip install --upgrade pycairo 'cairocffi>=0.8' PyGObject &&
+             python -c 'import gi; gi.require_version("Gtk", "3.0"); from gi.repository import Gtk' &&
+             echo 'PyGObject is available' ||
+             echo 'PyGObject is not available'
+
+          # There are no functioning wheels available for OSX 10.12 (as of
+          # Sept 2020) for either pyqt5 (there are only wheels for 10.13+) or
+          # pyside2 (the latest version (5.13.2) with 10.12 wheels has a
+          # fatal to us bug, it was fixed in 5.14.0 which has 10.13 wheels)
+          python -mpip install --upgrade pyqt5 &&
+            python -c 'import PyQt5.QtCore' &&
+            echo 'PyQt5 is available' ||
+            echo 'PyQt5 is not available'
+          python -mpip install --upgrade pyqt6 &&
+            python -c 'import PyQt6.QtCore' &&
+            echo 'PyQt6 is available' ||
+            echo 'PyQt6 is not available'
+          python -mpip install --upgrade pyside6 &&
+            python -c 'import PySide6.QtCore' &&
+            echo 'PySide6 is available' ||
+            echo 'PySide6 is not available'
+
+          python -mpip install --upgrade \
+            -f "https://extras.wxpython.org/wxPython4/extras/linux/gtk3/ubuntu-20.04" \
+            wxPython &&
+            python -c 'import wx' &&
+            echo 'wxPython is available' ||
+            echo 'wxPython is not available'
+
+      - name: Install Matplotlib
+        run: |
+          git describe
+
+          # All dependencies must have been pre-installed, so that the minver
+          # constraints are held.
+          python -m pip install --no-deps -e .
+
+      - name: Run pytest
+        run: |
+          xvfb-run -a python -mpytest -raR -n auto \
+            --maxfail=50 --timeout=300 --durations=25 --log-level=DEBUG \
+            -W error::UserWarning
```

Comment:
We actually would like to see the DeprecationWarning's as well (`-W error` without the specific UserWarning), but on Python 3.10 Numpy's testing import causes deprecation warnings due to the distutils future removal in Python 3.12. So, we would have to wait for Numpy's removal of distutils before we did that. Unless I'm missing a way to catch a specific string of deprecation warnings so we could ignore only that specific warning.

Reply:
Sure, you can be more specific with warnings filters: https://docs.python.org/3/library/warnings.html#describing-warning-filters

Reply:
Thanks for pointing that out. I tried a few things out and it was more complicated that I hoped.
See here for other discussion: https://bugs.python.org/issue43862
> While the Python API warnings.filterwarnings(action, message="", ...) uses the message as a regular expression, -W and PYTHONWARNINGS require to match *exactly* the *whole* message.

I got one match, and then another import failed, so not sure how many options/messages we want to match on if we do. I couldn't figure out how to use the "module" piece either, I thought that could refer to errors raise from `distutils` directly, or even by putting `numpy.testing` in there for where the warnings were arising from, but neither of those worked for me...

In summary, I'd leave it with the UserWarning -> Error for now and deal with DeprecationWarnings after Numpy fixes their distutils  imports.